File: LeetCode/424-longest-repeating-character-replacement/ex_424.java
public class Solution {
    public int characterReplacement(String s, int k) {
        int[] count = new int[26];
        int charCount = 0;
        int start = 0;
        int res = 0;
        int end = 0;
        while (end < s.length()) {
            count[s.charAt(end) - 'A']++;
            charCount = Math.max(charCount, count[s.charAt(end) - 'A']);
            end++;
            while (end - start - charCount > k) {
                count[s.charAt(start) - 'A']--;
                start++;
                for (int i : count) 
                    charCount = Math.max(charCount, i);
            }
            res = Math.max(res, end - start);
        }
        return res;
    }
}

File: LeetCode/424-longest-repeating-character-replacement/ex_424.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> count(26);
        int charCount = 0;
        int start = 0;
        int res = 0;
        for (int end = 0; end != s.size(); ++end) {
            ++count[s[end] - 'A'];
            charCount = max(charCount, count[s[end] - 'A']);
            while (end - start - charCount + 1 > k) {
                count[s[start] - 'A']--;
                start++;
                for (int &i : count) 
                    charCount = max(charCount, i);
            }
            res = max(res, end - start + 1);
        }
        return res;
    }
};